synapse/federation/transaction_queue.py

```diff
@@ -390,14 +400,20 @@ def send_edu(self, edu, key):
             edu (Edu): edu to send
             key (Any|None): clobbering key for this edu
         """
-        if key:
-            self.pending_edus_keyed_by_dest.setdefault(
-                edu.destination, {}
-            )[(edu.edu_type, key)] = edu
-        else:
-            self.pending_edus_by_dest.setdefault(edu.destination, []).append(edu)
+        dest = edu.destination
+        queue = self.pending_edus_by_dest.get(dest)
+        if not queue:
+            def tx_cb():
+                self._attempt_new_transaction(dest)
```

Comment:
I'd be tempted to give `PerDestinationQueue` the destination so that we don't have to keep creating new callbacks